buildroot-external/board/nabucasa/green/hassos-hook.sh

```diff
@@ -14,8 +14,7 @@ function hassos_pre_image() {
     # SPL
     create_spl_image
 
-    dd if="${BINARIES_DIR}/idbloader.img" of="${SPL_IMG}" conv=notrunc bs=512 seek=64
-    dd if="${BINARIES_DIR}/u-boot.itb" of="${SPL_IMG}" conv=notrunc bs=512 seek=16384
+    dd if="${BINARIES_DIR}/u-boot-rockchip.bin" of="${SPL_IMG}" conv=notrunc bs=512 seek=64
```

Comment:
Currently we have an extra partition marking/reserving the location of `u-boot.itb`. The second dd command used to reflect exactly this offset. But I guess the offset really is inherently tied to the  idbloader, so having them as one binary makes sense. Just wanted to note this.

Reply:
odroid-m1 hardcodes a partition for uboot but other boards dont? Anyway with default uboot config, it still ends up in the same place.

Reply:
Yes, ODROID-M1 is a bit special/the first one to do this.

Normally, partitions start at 1MB/2MB offset. Since U-Boot is at offset 8MB, it feels a bit weird having it there without the partition table knowing. So I've decided to give it an entry in the partition table. Since we use IDs everywhere it is safe to use a different layout for a particular board.

See: https://github.com/home-assistant/operating-system/blob/11.4/buildroot-external/scripts/hdd-image.sh#L165

This actually brings up a good point: We should definitely do this for ODROID-M1S too. And probably should have done it for Green too, but it feels a bit weird to change that now, since it would mean we'd end up having installations with different partition tables :thinking:

Reply:
I will make sure to do this on M1S then. 

I have noticed the offsets are a bit weird even still, with boot partition starting at 16M (from meta) +1M offset :confused:. But can't really change that now for existing boards!